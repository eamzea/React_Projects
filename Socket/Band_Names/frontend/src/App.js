import React, { useEffect, useState } from 'react';
import io from 'socket.io-client';
import { Col, Container, Row } from 'react-bootstrap';
import AddBand from './components/AddBand';
import BandList from './components/BandList';

const connectSocketServer = () => {
  const socket = io.connect('http://localhost:4000', {
    transports: ['websocket'],
  });

  return socket;
};

const App = () => {
  const [socket] = useState(connectSocketServer());
  const [isOnline, setIsOnline] = useState(false);
  const [bandsState, setBandsState] = useState([]);

  useEffect(() => {
    setIsOnline(socket.connected);
  }, [socket]);

  useEffect(() => {
    socket.on('connect', () => setIsOnline(true));
  }, [socket]);

  useEffect(() => {
    socket.on('disconnect', () => setIsOnline(false));
  }, [socket]);

  useEffect(() => {
    socket.on('current_bands', bands => {
      setBandsState(bands);
    });
  }, [socket]);

  const handleVote = id => socket.emit('vote_band', { id });

  const handleDelete = id => socket.emit('delete_band', { id });

  const handleChange = (name, id) => socket.emit('change_band', { name, id });

  const handleAdd = name => socket.emit('add_band', { name });

  return (
    <Container>
      <div className="alert">
        <p>
          Service status:
          {isOnline ? (
            <span className="text-success"> Online</span>
          ) : (
            <span className="text-danger"> Offline</span>
          )}
        </p>
      </div>
      <h1>Band Names</h1>
      <hr />
      <Row>
        <Col xs="8">
          <BandList
            data={bandsState}
            vote={handleVote}
            deleteFn={handleDelete}
            changeName={handleChange}
          />
        </Col>
        <Col xs="4">
          <AddBand add={handleAdd} />
        </Col>
      </Row>
    </Container>
  );
};

export default App;
