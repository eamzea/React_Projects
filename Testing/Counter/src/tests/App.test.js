import React from "react";
// import { render } from "@testing-library/react"; JEST EXTEND
import App from "../App";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

describe("Pruebas en <App/>", () => {
  // test('Debe mostrar el mensaje "Hola, Soy Goku"', () => {
  //   const saludo = "Hola, Soy Goku";

  //   const { getByText } = render(<App saludo={saludo} />);

  //   expect(getByText(saludo)).toBeInTheDocument();
  // JEST EXTEND
  // });
  test("Debe de mostrar <App/> correctamente", () => {
    const saludo = "Hola, Soy Goku";
    const wrapper = shallow(<App saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar subtÃ­tulo enviado por props", () => {
    const saludo = "Hola, Soy Goku";
    const subtitulo = "Soy un subtitulo";
    const wrapper = shallow(<App saludo={saludo} subtitulo={subtitulo} />);
    const textoParrafo = wrapper.find("p").text();

    expect(textoParrafo).toBe(subtitulo);
  });
});
