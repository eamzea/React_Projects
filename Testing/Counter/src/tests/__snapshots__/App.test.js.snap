// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App/> Debe de mostrar <App/> correctamente 1`] = `
<div
  className="App"
>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    Soy un subt√≠tulo
  </p>
</div>
`;
